gradle/mockito-core/osgi.gradle

```diff
@@ -27,88 +18,8 @@ jar {
                 'org.mockito.*'
         ].join(','),
         '-removeheaders': 'Private-Package',
-        'Automatic-Module-Name': 'org.mockito'
+        'Automatic-Module-Name': 'org.mockito',
+        '-noextraheaders': 'true',
+        '-export-apiguardian': 'org.mockito.internal.*'
     )
 }
-
-jar.doLast {
-    JarFile originalJar = new JarFile(jar.archivePath)
-
-    new RemoveOsgiLastModifiedHeader(originalJar)
-        .transform()
-        .renameTo jar.archivePath
-}
-
-
-/*
- * The OSGi plugin in Gradle 5.3 has a known issue (https://github.com/gradle/gradle/issues/6187) where it embeds
- * a Manifest entry "Bnd-LastModified" (set to the last modified time of the build directory) even if you ask it not to
- * using `-noextraheaders` or `-removeheaders`.
- *
- * It cannot be pinned or removed, and is a source of non-determinism in the build.
- *
- * This class addresses the problem by rewriting the JAR and removing that entry from the Manifest. A side-effect of this
- * is having to set `lastModifiedTime` for each entry to a fixed value, else this would also introduce non-determinism.
- *
- * The fixed value is the same as the default value for Gradle's fixed timestamps, and the rationale behind it is detailed
- * in https://github.com/gradle/gradle/blob/58538513a3bff3b7015718976fe1304e23f40694/subprojects/core/src/main/java/org/gradle/api/internal/file/archive/ZipCopyAction.java#L42-L57
- */
-
-class RemoveOsgiLastModifiedHeader {
```

Comment:
It's not clear to me why this part was removed and what changed it. Could you please add a comment to the functionality that replaces it to explain why we need it? (E.g. reproducible jars) And could you update your commit message to explain the change that was made.

Reply:
bnd has this feature builtin! I enabled it by adding `-noextraheaders: true` rendering this code obsolete.

Reply:
if you diff the results before and after (which I did to double check) you will see what I mean :)